deepmd/dpmodel/utils/nlist.py

```
@@ -163,20 +163,21 @@ def nlist_distinguish_types(
     ret_nlist = []
-    tmp_atype = xp.tile(atype[:, None], [1, nloc, 1])
+    tmp_atype = xp.tile(atype[:, None, :], (1, nloc, 1))
     mask = nlist == -1
-    tnlist_0 = nlist.copy()
-    tnlist_0[mask] = 0
-    tnlist = xp_take_along_axis(tmp_atype, tnlist_0, axis=2).squeeze()
-    tnlist = xp.where(mask, -1, tnlist)
+    tnlist_0 = xp.where(mask, xp.zeros_like(nlist), nlist)
+    tnlist = xp_take_along_axis(tmp_atype, tnlist_0, axis=2)
+    tnlist = xp.where(mask, xp.full_like(tnlist, -1), tnlist)
     snsel = tnlist.shape[2]
```

Ruff

Local variable `snsel` is assigned to but never used

Remove assignment to unused variable `snsel`

(F841)